frontend/server/src/Controllers/User.php

```diff
@@ -1382,22 +1382,30 @@ public static function apiExtraInformation(\OmegaUp\Request $r): array {
      */
     public static function apiCoderOfTheMonth(\OmegaUp\Request $r) {
         $date = !empty($r['date']) ? strval($r['date']) : null;
+        $category = !empty($r['category']) ? strval($r['category']) : 'all';
```

Comment:
```suggestion
    const ALLOWED_CODER_OF_THE_MONTH_CATEGORIES = [
        'all', 'female',
    ];
    ...
        \OmegaUp\Validators::validateOptionalInEnum(
            $r['category'],
            'category',
            \OmegaUp\Controllers\User::ALLOWED_CODER_OF_THE_MONTH_CATEGORIES
        );
        $category = !empty($r['category']) ? $r['category'] : 'all';
```

igual en todos los demás lugares donde usas `$r['category']`

Reply:
@lhchavez la constante va como variable global?

Reply:
@lhchavez Para que sirve el validateOptionalInEnum?

Reply:
> @lhchavez la constante va como variable global?

como constante estática dentro de la clase.

Reply:
> @lhchavez Para que sirve el validateOptionalInEnum?

para que si alguien manda algo que no está dentro del conjunto de opciones válidas, se le notifica que mandó un parámetro inválido.